### PR TITLE
fabtest: Add fi_rdm_bw test

### PR DESCRIPTION
**Pull request by shijin-aws**
This is mostly a copy of fi_rdm_tagged_bw but test with untagged msg.

### CODE REVIEW

**Comment by shijin-aws** [comment]
@j-xiong @zachdworkin is intel CI failure related?

**Comment by shijin-aws** [comment]
OK I found log for the windows test

```
projects\libfabric\fabtests\fabtests.vcxproj]
  NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\bin\HostX86\x64\cl.EXE"' : return code '0x2' [C:\projects\libfabric\fabtests\fabtests.vcxproj]
  rdm_bw.obj : error LNK2019: unresolved external symbol bandwidth referenced in function run [C:\projects\libfabric\fabtests\fabtests.vcxproj]
  rdm_bw.obj : error LNK2001: unresolved external symbol optarg [C:\projects\libfabric\fabtests\fabtests.vcxproj]
  rdm_bw.obj : error LNK2001: unresolved external symbol optind [C:\projects\libfabric\fabtests\fabtests.vcxproj]
  rdm_bw.obj : error LNK2001: unresolved external symbol test_size [C:\projects\libfabric\fabtests\fabtests.vcxproj]
  rdm_bw.obj : error LNK2001: unresolved external symbol test_cnt [C:\projects\libfabric\fabtests\fabtests.vcxproj]
  rdm_bw.obj : error LNK2001: unresolved external symbol fi [C:\projects\libfabric\fabtests\fabtests.vcxproj]
  rdm_bw.obj : error LNK2001: unresolved external symbol hints [C:\projects\libfabric\fabtests\fabtests.vcxproj]
  rdm_bw.obj : error LNK2001: unresolved external symbol test_name [C:\projects\libfabric\fabtests\fabtests.vcxproj]
  rdm_bw.obj : error LNK2001: unresolved external symbol opts [C:\projects\libfabric\fabtests\fabtests.vcxproj]
  rdm_bw.obj : error LNK2001: unresolved external symbol lopt_idx [C:\projects\libfabric\fabtests\fabtests.vcxproj]
  rdm_bw.obj : error LNK2001: unresolved external symbol long_opts [C:\projects\libfabric\fabtests\fabtests.vc
```

**Comment by zachdworkin** [comment]
They look unrelated but I am re-running just the stages that failed to be sure.

**Comment by shijin-aws** [comment]
@zachdworkin is Intel CI failure related?

**Comment by zachdworkin** [comment]
@shijin-aws it is unclear. I am going to re-run just those stages to see if they just had a blip.
There were two errors:
UCX: Failed "fi_rdm_tagged_peek" - this one is known
TCP-io_uring: Failed fi_multi_recv due to timeout
- name:   fi_multi_recv -e rdm -p "tcp"
   timestamp: 20240618-184847+0000
   result: Fail
   time:   302
   server_cmd:  /path-to-libfabric/fabtests/bin/fi_multi_recv -e rdm -p "tcp"   -s node1-iface
   server_stdout: |
   client_cmd:  /path-to-libfabric/fabtests/bin/fi_multi_recv -e rdm -p "tcp"   -s node2-iface node1-iface
   client_stdout: |

**Comment by zachdworkin** [comment]
@shijin-aws It looks like the tcp-iouring failure was just a blip. I am seeing the ucx fi_rdm_tagged_peek test consitently failing but it is not from your changes. I have other jobs running that also see that test consistently fail and I will look into it and open a PR for a fix if I can find one.

File: fabtests/benchmarks/rdm_bw.c [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+			ft_parse_benchmark_opts(op, optarg);
+			ft_parseinfo(op, optarg, hints, &opts);
+			ft_parsecsopts(op, optarg, &opts);
+			break;
+		case 'U':
+			hints->tx_attr->op_flags |= FI_DELIVERY_COMPLETE;
+			break;
+		case '?':
+		case 'h':
+			ft_csusage(argv[0], "Bandwidth test for RDM endpoints using tagged messages.");
```
Comment (j-xiong) [inline fabtests/benchmarks/rdm_bw.c:70]: The help text still says `tagged` message.

Comment (shijin-aws) [reply fabtests/benchmarks/rdm_bw.c]: Fixed

**Review by j-xiong — APPROVED** [review]

**Comment by zachdworkin** [comment]
@shijin-aws Here is your CI output. This is the failure that we occasionally see. I currently think it is a race condition and unrelated to your changes.
- name:   fi_rdm_tagged_peek -p "ucx"
   timestamp: 20240624-185709+0000
   result: Fail
   time:   2
   server_cmd:  /path_to_fabtests_install/bin/fi_rdm_tagged_peek -p "ucx"   -E
   server_stdout: |
     Sending 10 tagged messages
     Waiting for messages to complete
     [n1:3394645:0:3394645] Caught signal 11 (Segmentation fault: address not mapped to object at address 0x8)
     ==== backtrace (tid:3394645) ====
      0 0x0000000000012cf0 __funlockfile()  :0
      1 0x0000000000033210 ucp_ep_destroy_base()  ???:0
      2 0x000000000004b3ee ucp_worker_discard_uct_ep_progress()  ???:0
      3 0x000000000004b4b5 ucp_worker_destroy()  ???:0
      4 0x00000000000ca4ca ucx_ep_close()  ucx_ep.c:0
      5 0x0000000000404081 fi_close()  /path_to_libfabric_source/include/rdma/fabric.h:632
      6 0x0000000000404081 ft_close_fids()  /path_to_libfabric_source/libfabric/fabtests/common/shared.c:1776
      7 0x0000000000404b6a ft_free_res()  /path_to_libfabric_source/fabtests/common/shared.c:1846
      8 0x0000000000401bfa main()  /path_to_libfabric_source/fabtests/functional/rdm_tagged_peek.c:364
      9 0x0000000000401bfa main()  /path_to_libfabric_source/fabtests/functional/rdm_tagged_peek.c:365
     10 0x000000000003ad85 __libc_start_main()  ???:0
     11 0x000000000040203e _start()  ???:0
     =================================
   client_cmd:  /path_to_fabtests_install/bin/fi_rdm_tagged_peek -p "ucx"   -E n2-iface
   client_stdout: |
     Peek for a bad msg
     Peek w/ claim for a bad msg
     Peek msg 1
     Receive msg 1
     Peek w/ claim msg 2
     Receive claimed msg 2
     Peek & discard msg 3
     Checking to see if msg 3 was discarded
     Peek w/ claim msg 4
     Claim and discard msg 4
     Receive msg 5
     Receive msg 6
     Receive msg 10
     Receive msg 9
     Receive msg 8
     Receive msg 7

**Comment by shijin-aws** [comment]
@zachdworkin thx, merging...

**Pull request merged**